component/receiver.go

```diff
@@ -63,6 +63,9 @@ type ReceiverCreateParams struct {
 
 	// ApplicationStartInfo can be used by components for informational purposes
 	ApplicationStartInfo ApplicationStartInfo
+
```

Comment:
@jpkrohling @tigrannajaryan This commit removes the global registry and passes the built extensions into the ReceiverCreateParams to be used eventually by the configgrpc code. I can clean this up more, but wanted you to take a look before I keep going.